security/tpm/test/_tpm_test.js

```diff
@@ -102,12 +182,18 @@ describe('tpm', function () {
 
     it('must call activateIdentityKey first', function(done) {
       /*Tests_SRS_NODE_TPM_SECURITY_CLIENT_06_013: [If `signWithIdentity` is invoked without a previous successful invocation of `activateIdentityKey`, the callback will be invoked with `err` of `InvalidOperationError`.] */
-      var client = new TpmSecurityClient('registration', fakeSimpleTpmClient);
-      var createPersistStub = sinon.stub(client,'_createPersistentPrimary');
-      var readPersistStub = sinon.stub(client,'_readPersistentPrimary');
-      createPersistStub.withArgs('EK').callsArgWith(4, null, TpmSecurityClient._ekTemplate);
-      createPersistStub.withArgs('SRK').callsArgWith(4, null, TpmSecurityClient._srkTemplate);
-      readPersistStub.withArgs('IDENTITY').callsArgWith(2, null, null);
+      var tpm = new tss.Tpm();
+      var connectStub = sinon.stub(tpm, 'connect');
+      connectStub.callsArgWith(0);
+      var client = new TpmSecurityClient(undefined, tpm);
+      var readPublicStub = sinon.stub(tpm, 'ReadPublic');
+      var lastResponseStub = sinon.stub(tpm, 'getLastResponseCode');
+      lastResponseStub.onCall(0).returns(tss.TPM_RC.SUCCESS);
+      lastResponseStub.onCall(1).returns(tss.TPM_RC.SUCCESS);
+      lastResponseStub.onCall(2).returns(tss.TPM_RC.BAD_TAG);
+      readPublicStub.withArgs(TpmSecurityClient._ekPersistentHandle).callsArgWith(1, fakeEkKey);
+      readPublicStub.withArgs(TpmSecurityClient._srkPersistentHandle).callsArgWith(1, fakeSrkKey);
+      readPublicStub.withArgs(TpmSecurityClient._idKeyPersistentHandle).callsArgWith(1, fakeIdKey);
       client.signWithIdentity([1], (err, signResult) => {
         assert.instanceOf(err, common.errors.InvalidOperationError, 'should indicate invalid operation if active not called first.');
```

Comment:
>active [](start = 106, length = 6)

typo? if activateIdentityKey is not called first?

Reply:
It was my way of being terse.  Added the two additional qualifications.

---
In reply to: [167386022](https://github.com/Azure/azure-iot-sdk-node/pull/225#discussion_r167386022) [](ancestors = 167386022)